### DISCUSSION TITLE
Allow Blade to render enums

### DISCUSSION THREAD

**Discussion by brunoalod** [laravel/framework#44440] [Ideas] [answered]
Follow up of #44431.

Echoing a enum in Blade causes an exception. I was thinking it might be a nice feature to echo either its name or its value according to the type of enum.

```PHP
return view('my-view', [
    'enum' => MyEnum::myValue,
]);
```

```BLADE
// view
{{ $enum }}
```
```
htmlspecialchars(): Argument #1 ($string) must be of type string, MyEnum given
```

**Comment by henzeb** [comment]
my package can do that for you in the meantime: https://github.com/henzeb/enumhancer

**Comment by nunomaduro** [comment, marked as answer]
Thank you - I will take a look if is possible to fix this.

**Reply by brunoalod** [reply]
Thanks for looking into it! I can attempt a PR if needed.

**Comment by parallels999** [comment]
Read other discussions
https://github.com/laravel/framework/discussions/39895#discussioncomment-2830097

**Reply by brunoalod** [reply]
A PR was sent, and merged.
